### DISCUSSION TITLE
Relation manager actions model

### DISCUSSION THREAD

**Discussion by kurucu** [filamentphp/filament#8239] [Help] [answered]
Hi all,

I want to fire an event after attaching and detaching from a resource, and need to include the model in the event. I can't find any guidance on this - any ideas?

Can I use `$this->ownerModel` everywhere for example?

Thanks!

**Comment by leandrocfe** [comment]
I think you can use `after()` method
```php
AttachAction::make()
    ->after(function ($record) {
        
        //$record
        //$this->record
        //$this->ownerRecord

    })
```

**Reply by kurucu** [reply]
Unfortunately none of those worked for me. $record wasn't found, $this->record and $this->ownerRecord led to the same error, that ownerRecord cannot be accessed before initialisation.

**Comment by kurucu** [comment, marked as answer]
After a lot of trial and error, I found that this worked for both attach and detach events:

```php
AttachAction::make()
    //...
    ->after(function($livewire){
        // use $livewire->ownerRecord here
    })
```
